packages/babel-helper-validator-option/src/validator.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+import { findSuggestion } from "./find-suggestion.js";
+
+export class OptionValidator {
+  declare descriptor: string;
+  constructor(descriptor: string) {
+    this.descriptor = descriptor;
+  }
+
+  /**
+   * Validate if the given `options` follow the name of keys defined in the `TopLevelOptionShape`
+   *
+   * @param {Object} options
+   * @param {Object} TopLevelOptionShape
+   *   An object with all the valid key names that `options` should be allowed to have
+   *   The property values of `TopLevelOptionShape` can be arbitrary
+   * @memberof OptionValidator
+   */
+  validateTopLevelOptions(options: Object, TopLevelOptionShape: Object): void {
+    const validOptionNames = Object.keys(TopLevelOptionShape);
+    for (const option of Object.keys(options)) {
+      if (!validOptionNames.includes(option)) {
+        throw new Error(
+          this.formatMessage(`'${option}' is not a valid top-level option.
+- Maybe you are meant to use '${findSuggestion(option, validOptionNames)}'?`),
```

Comment:
```suggestion
- Maybe you meant to use '${findSuggestion(option, validOptionNames)}'?`),
```

Reply:
https://www.merriam-webster.com/dictionary/meant%20to%20%28be%2Fdo%20something%29 It seems that `be meant to` is correct. I can rephrase into `intended to` if you prefer.

Reply:
Or even just "Did you mean ...?"